ci/cleanup-pr-previews

```diff
@@ -0,0 +1,222 @@
+            FOLDERS_TO_REMOVE+=("$folder")
+            ;;
+        "not_found")
+            echo -e "${RED}[REMOVE]${NC} PR #${PR_NUMBER} not found (may have been deleted)"
+            FOLDERS_TO_REMOVE+=("$folder")
+            ;;
+        *)
+            echo -e "${YELLOW}[UNKNOWN]${NC} PR #${PR_NUMBER} has unexpected status: ${PR_STATUS}"
+            ;;
+    esac
+done <<< "$PR_FOLDERS"
+
+# Summary
+echo ""
+echo -e "${YELLOW}[SUMMARY]${NC}"
+echo "Total PR preview folders: ${TOTAL_FOLDERS}"
+echo "Open PRs: ${OPEN_PRS}"
+echo "Folders to remove: ${#FOLDERS_TO_REMOVE[@]}"
+
+if [[ ${#FOLDERS_TO_REMOVE[@]} -eq 0 ]]; then
+    echo -e "${GREEN}[INFO]${NC} No cleanup needed - all preview folders correspond to open PRs"
+    exit 0
+fi
+
+# List folders to be removed
+echo ""
+echo -e "${YELLOW}[FOLDERS TO REMOVE]${NC}"
+for folder in "${FOLDERS_TO_REMOVE[@]}"; do
+    pr_num="${folder#pr-}"
+    echo "  - $folder (PR #${pr_num})"
+done
+
+# Perform cleanup or show what would be done
+echo ""
+if [[ "$DRY_RUN" == "true" ]]; then
+    echo -e "${YELLOW}[DRY RUN]${NC} Would remove ${#FOLDERS_TO_REMOVE[@]} folders (use dry-run=false to actually remove)"
+else
+    echo -e "${RED}[CLEANUP]${NC} Proceeding to remove ${#FOLDERS_TO_REMOVE[@]} folders..."
+    
+    # Clone gh-pages branch to a temporary directory
+    TEMP_DIR=$(mktemp -d)
+    trap 'rm -rf "$TEMP_DIR"' EXIT
+    
+    echo -e "${YELLOW}[INFO]${NC} Cloning gh-pages branch to temporary directory..."
+    git clone --depth 1 --branch gh-pages "https://github.com/${REPOSITORY}.git" "$TEMP_DIR" >/dev/null 2>&1
+    
+    cd "$TEMP_DIR"
+    
+    # Configure git for the cleanup commit
+    git config user.name "cuda-python-bot"
+    git config user.email "cuda-python-bot@users.noreply.github.com"
+    
+    # Remove each folder
+    REMOVED_COUNT=0
+    for folder in "${FOLDERS_TO_REMOVE[@]}"; do
+        pr_num="${folder#pr-}"
+        folder_path="docs/pr-preview/$folder"
+        
+        if [[ -d "$folder_path" ]]; then
+            echo -e "${YELLOW}[REMOVE]${NC} Removing $folder_path"
+            rm -rf "$folder_path"
+            git add "$folder_path"
+            REMOVED_COUNT=$((REMOVED_COUNT + 1))
+        else
+            echo -e "${YELLOW}[SKIP]${NC} Folder $folder_path not found locally"
+        fi
+    done
+    
+    if [[ $REMOVED_COUNT -gt 0 ]]; then
+        # Commit and push changes
+        commit_message="Clean up PR preview folders for ${REMOVED_COUNT} closed/merged PRs
+
+Removed preview folders for the following PRs:
+$(printf '%s\n' "${FOLDERS_TO_REMOVE[@]}" | sed 's/^pr-/- PR #/' | head -20)
+$(if [[ ${#FOLDERS_TO_REMOVE[@]} -gt 20 ]]; then echo "... and $((${#FOLDERS_TO_REMOVE[@]} - 20)) more"; fi)"
+        
+        echo -e "${YELLOW}[INFO]${NC} Committing changes..."
+        git commit -m "$commit_message"
+        
+        echo -e "${YELLOW}[INFO]${NC} Pushing to gh-pages branch..."
+        git push origin gh-pages
```

Comment:
I don't want to do these two steps by default. Please add another option to allow these steps (and default is false). This is very unsafe if I want to verify the diff locally before pushing.